src/components/2FA/index.tsx

```diff
@@ -0,0 +1,132 @@
+import { XMarkIcon } from '@heroicons/react/24/solid'
+import { FC, useEffect, useState } from 'react'
+import { Heading2 } from 'styles/typography/styles'
+import { FunctionWithParams, SuccessAuthResponse } from 'types'
+import { noopFunction } from 'utils/noop'
+
+import { LittleLoader } from 'components/loaders/littleLoader'
+
+import { useConfirmUser2FAMutation } from './api'
+import { AUTH_FAILURE_MESSAGE, AUTH_SYMBOLS_ARRAY, BACKSPACE_CODE, ENTER_CODE } from './const'
+import {
+  QrCodeWrapper,
+  User2FAButton,
+  User2FACloseModalButton,
+  User2FAComponentBlock,
+  User2FAErrorSpan,
+  User2FAInput,
+  User2FASuccessSpan,
+  User2FAWrapper
+} from './styles'
+
+type Props = {
+  qrcode: string
+  userId: string
+  handleCloseModal: VoidFunction
+  onSuccessCallback: FunctionWithParams<SuccessAuthResponse>
+  onErrorCallback?: FunctionWithParams<string>
+}
+
+export const User2FAComponent: FC<Props> = ({
+  qrcode,
+  userId,
+  onSuccessCallback,
+  handleCloseModal,
+  onErrorCallback = noopFunction
+}) => {
+  const [attemptCount, setAttempCount] = useState(3)
+  const [code, setCode] = useState('')
+  const [submit2faData, { isLoading, isError, isSuccess, data }] = useConfirmUser2FAMutation()
+
+  useEffect(() => {
+    if (attemptCount <= 0) {
+      handleCloseModal()
+      onErrorCallback(AUTH_FAILURE_MESSAGE)
+    }
+  }, [attemptCount, onErrorCallback, handleCloseModal])
+
+  useEffect(() => {
+    if (isError) {
+      setAttempCount((prev) => --prev)
+    }
+  }, [isError])
+
+  useEffect(() => {
+    let timerId: NodeJS.Timeout
+
+    if (isSuccess) {
+      timerId = setTimeout(() => {
+        onSuccessCallback(data as SuccessAuthResponse)
+      }, 2000)
+    }
+
+    return () => {
+      if (timerId) {
+        clearTimeout(timerId)
+      }
+    }
+  }, [isSuccess, onSuccessCallback, data])
+
+  const handleSubmit = async () => submit2faData({ code, userId })
+
+  const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    if (code.length < 6) {
+      setCode(e.target.value)
+    }
+  }
+
+  const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e) {
+      if (AUTH_SYMBOLS_ARRAY.includes(e.key)) {
+        e.preventDefault()
+      }
+
+      if (code.length === 6 && e.code === BACKSPACE_CODE) {
+        setCode((prev) => prev.slice(0, 5))
+        e.preventDefault()
+      }
```

Comment:
Does it works when you tap on backspace when len is not equal 6?

Reply:
This code was updated on regularExp, example 
<pre>
 const handleChange = (e: React.ChangeEvent<HTMLInputElement>) => {
    if (/^[0-9]{0,6}$/.test(e.target.value) || !e.target.value) {
      setCode(e.target.value)
    }

    e.preventDefault()
  }
  </pre>